tests/compile-fail/needless_borrow.rs

```diff
@@ -33,3 +33,18 @@ trait Trait {}
 impl<'a> Trait for &'a str {}
 
 fn h(_: &Trait) {}
+
+#[allow(dead_code, unused_mut, match_ref_pats, unused_variables)]
+fn issue_1432() {
+    let mut s = String::new();
+    match & &mut s {
+        & ref a => {}
```

Comment:
oh, that was fine, I meant the `&ref` pattern, I added an extra issue for it: https://github.com/Manishearth/rust-clippy/issues/1434

The new examples are great!